src/vs/workbench/contrib/preferences/browser/settingsTreeModels.ts

```diff
@@ -288,6 +289,32 @@ export class SettingsTreeSettingElement extends SettingsTreeElement {
 
 		return Array.from(extensionFilters).some(extensionId => extensionId.toLowerCase() === this.setting.extensionInfo!.id.toLowerCase());
 	}
+
+	matchesAnyFeature(featureFilters?: Set<string>): boolean {
+		const features = tocData.children!.filter(child => child.id === 'features')[0].children!.map(feature => {
+			return {
+				name: feature.id.substring(9),
+				values: feature.settings!.map(setting => setting.toString().substring(0, setting.toString().length - 2))
+			};
+		});
+
+		if (!featureFilters || !featureFilters.size) {
+			return true;
+		}
+		const result = Array.from(featureFilters).map(filter => {
+			const foundFeature = features.find(item => item.name === filter);
+			if (foundFeature) {
+				if (foundFeature.values.length === 1) {
+					return Array.from(featureFilters).some((feature: string) => this.setting.key.toLowerCase().startsWith(feature));
```

Comment:
This should be evaluating the settings glob patterns, not using `startsWith`. See `resolveSettingsTree` in `SettingsTree.ts` for the code that does this. You can extract part of that as a helper if it makes sense.